src/header.rs

```diff
@@ -231,6 +231,31 @@ impl NiftiHeader {
         parse_header_1(input)
     }
 
+    /// Retrieve and validate the dimensions of the volume. Unlike how NIfTI-1
+    /// stores dimensions, the returned slice does not include `dim[0]` and is
+    /// clipped to the effective number of dimensions.
+    /// 
+    /// # Error
+    /// 
+    /// `NiftiError::` if `dim[0]` does not represent a valid dimensionality.
+    pub fn dim(&self) -> Result<&[u16]> {
+        let ndim = self.dimensionality()?;
+        Ok(&self.dim[1..ndim + 1])
+    }
+
+    /// Retrieve and validate the number of dimensions of the volume. This is
+    /// `dim[0]` after the necessary byte order conversions.
+    /// 
+    /// # Error
+    /// 
+    /// `NiftiError::` if `dim[0]` does not represent a valid dimensionality.
+    pub fn dimensionality(&self) -> Result <usize> {
+        if self.dim[0] > 7 {
```

Comment:
Is 0 a valid dimensionality?

Reply:
Great question, I don't know. Might be worth looking into this.